test/test-ExperimentProvider.js

```diff
@@ -0,0 +1,105 @@
+"use strict";
+
+const test = require("sdk/test");
+const {ExperimentProvider} = require("lib/ExperimentProvider");
+const {ActivityStreams} = require("lib/ActivityStreams");
+
+const DEFAULT_CLIENT_ID = "foo";
+const DEFAULT_TEST_EXPERIMENTS = {
+  foo: {
+    id: "foo_01",
+    name: "Foo Test",
+    description: "A test about foo",
+    control: {
+      value: 42,
+      description: "Foo is 42 by default"
+    },
+    variant: {
+      value: 84,
+      threshold: 0.5,
+      description: "Twice the foo"
+    }
+  }
+};
+
+let experimentProvider;
+
+// // Utility for generating an experiment based on a valid definition
+// function createExperiment(details = {}) {
+//   return Object.assign({}, DEFAULT_TEST_EXPERIMENTS.foo, details);
+// }
+
+function setup(clientID = DEFAULT_CLIENT_ID, experiments = DEFAULT_TEST_EXPERIMENTS) {
+  return experimentProvider = new ExperimentProvider(clientID, experiments);
+}
+
+exports["test ExperimentProvider"] = assert => {
+  setup();
+  assert.ok(experimentProvider._clientID, "should have a .clientID property");
+  assert.ok(experimentProvider._rng, "should have a ._rng property");
+  assert.ok(experimentProvider._data, "should have a ._data property");
+};
+
+exports["test ExperimentProvider._rng"] = assert => {
+  setup("foo");
+  assert.equal(Math.round(experimentProvider._rng() * 100) / 100, 0.73, "seed foo should generate 0.73");
+  setup("012j");
+  assert.equal(Math.round(experimentProvider._rng() * 100) / 100, 0.08, "seed 012j should generate 0.08");
+};
+
+exports["test ExperimentProvider.data"] = assert => {
+  setup("baz");
+  assert.equal(experimentProvider.data, experimentProvider._data, ".data should return return this._data");
+  assert.deepEqual(experimentProvider.data, {
+    foo: {id: "foo_01", value: 42, inExperiment: false}
+  }, "clientID 'baz' should result in control being picked");
+
+  setup("012j");
+  assert.deepEqual(experimentProvider.data, {
+    foo: {id: "foo_01", value: 84, inExperiment: true}
+  }, "clientID '012j' should result in variant being picked");
+};
+
+// exports["test ExperimentProvider.validateExperiment"] = assert => {
```

Comment:
`assert.throws` seems to intermittently fuck up the tests, I get this output sometimes:

```
console.error: activity-streams:
  warnings and/or errors were logged.

26 of 27 tests passed.
There were test failures...

The following tests failed:


```

i.e., it causes a failure, but no tests actually failed?